tests/unittests/client/test_experiment_client.py

```diff
@@ -969,3 +974,36 @@ def foo(a, b):
             assert len(params)
             assert "x" in params["a"]
             assert "y" in params["b"]
+
+    def test_parallel(self, monkeypatch):
```

Comment:
It's good to have a test with a mocked call like this, but I think there should also be a test where the optimization is actually done to make sure race conditions are handled properly.